rust/lance-io/src/utils/tracking_store.rs

```diff
@@ -191,22 +229,28 @@ impl IoTrackingStore {
         let mut stats = self.stats.lock().unwrap();
         stats.read_iops += 1;
         stats.read_bytes += num_bytes;
+        #[cfg(feature = "test-util")]
         stats.requests.push(IoRequestRecord {
             method,
             path,
             range,
         });
+        #[cfg(not(feature = "test-util"))]
+        let _ = (method, path, range); // Suppress unused variable warnings
     }
 
     fn record_write(&self, method: &'static str, path: Path, num_bytes: u64) {
         let mut stats = self.stats.lock().unwrap();
         stats.write_iops += 1;
         stats.write_bytes += num_bytes;
+        #[cfg(feature = "test-util")]
         stats.requests.push(IoRequestRecord {
             method,
             path,
             range: None,
         });
+        #[cfg(not(feature = "test-util"))]
+        let _ = (method, path); // Suppress unused variable warnings
```

Comment:
What are we feature gating here?  Tracking of every request's path / method / range vs. just tracking the counts?

Reply:
Yeah with `test-util` enabled, we will track a list of all requests made in the IO stats. This makes it much easier to debug a failing test.

But for normal usage, keeping track of those will just accumulate a lot of data for now reason.

This particular gate is just because `method` and `path` are only used for the request tracking, so added a line to suppress the unused variable warning.